### PR TITLE
feat(cb2-8712): dates don't error till user is done typing

### PR DESCRIPTION
**Pull request by naathanbrown**
## Ticket title

Dates don't show error messages until a user has touched all boxes. If they miss a box totally, errors will now show locally on submission.

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

**Review by shivangidas — DISMISSED** [review]

File: src/app/forms/components/date/date.component.ts [hunk 1]
```diff
@@ -44,20 +43,26 @@ export class DateComponent extends BaseControlComponent implements OnInit, OnDes
...
+  protected formSubmitted? = false;
 
   public day?: number;
   public month?: number;
   public year?: number;
   public hour?: number;
   public minute?: number;
 
-  constructor(injector: Injector, changeDetectorRef: ChangeDetectorRef) {
+  constructor(injector: Injector, changeDetectorRef: ChangeDetectorRef, public globalErrorService: GlobalErrorService) {
```
Comment (Daniel-Searle) [inline src/app/forms/components/date/date.component.ts:54]: Is public needed here?

File: src/app/forms/components/date/date.component.spec.ts [hunk 2]
```diff
@@ -24,17 +27,23 @@ class HostComponent {
...
+      imports: [FormsModule, ReactiveFormsModule],
+      providers: [GlobalErrorService, provideMockStore({ initialState: initialAppState })]
     }).compileComponents();
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(HostComponent);
     component = fixture.componentInstance;
+    store = TestBed.inject(MockStore);
+    //store.overrideSelector(globalErrorState, expectedErrors);
```
Comment (Daniel-Searle) [inline src/app/forms/components/date/date.component.spec.ts:45]: Is this still needed?

Comment (naathanbrown) [reply src/app/forms/components/date/date.component.spec.ts:45]: no

File: src/app/forms/components/date/date.component.ts (hunk 1 above)
Comment (naathanbrown) [reply src/app/forms/components/date/date.component.ts:54]: I think so?

**Review by Daniel-Searle — APPROVED** [review]

**Review by shivangidas — APPROVED** [review]

**Pull request merged**